core/src/main/java/tech/tablesaw/io/ReadOptions.java

```diff
@@ -204,6 +223,10 @@ public DateTimeFormatter dateFormatter() {
     protected int maxCharsPerColumn = 4096;
     protected boolean ignoreZeroDecimal = DEFAULT_IGNORE_ZERO_DECIMAL;
     private boolean allowDuplicateColumnNames = false;
+    protected ColumnType[] columnTypes;
```

Comment:
I think we could make this cleaner than having four different ways of deciding. We could have a single interface that's responsible for making the decision and you could have a different implementation based on each of the user options. Then we would only need one variable here